cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -359,6 +425,18 @@ const FunctionDoc div_checked_doc{
      "integer overflow is encountered."),
     {"dividend", "divisor"}};
 
+const FunctionDoc pow_doc{
+    "Raise arguments to power element-wise",
+    ("Integer to negative integer power returns an error. However, integer overflow\n"
+     "wraps around. Floating poing power follows std::pow() behaviour.\n"),
```

Comment:
I think ideally the behaviour is actually spelled out, instead of referring to `std::pow()`. Those docs are also used for python docstrings, and the typical python user will not even know what `std::pow()` is referring to.

Reply:
And I would maybe also add the "If either base or exponent is null the result will be null."

Reply:
Thanks @jorisvandenbossche

Reply:
Will add null behaviour.
For std::pow(), it looks too complicated: https://en.cppreference.com/w/cpp/numeric/math/pow,
maybe we can pick some common cases.

Reply:
Yeah, that's a lot (I didn't actually check it). But we can indeed pick the most important cases. 

I was mostly thinking about the special cases around 0 and 1:

> pow(+1, exp) returns 1 for any exp, even when exp is NaN
> pow(base, ±0) returns 1 for any base, even when base is NaN 

But on the other hand, I see that also `np.power` docstring doesn't actually mention special cases

Reply:
Guess that's why everyone hates floating points :)
I would prefer dropping descriptions about floating points. I don't think I can explain it clearly with my poor English.

Reply:
Sounds good. It's hard to draw a line what to mention and what not, anyway. It can always still be improved later on.

Reply:
We could add the cpp docs link the docstring?